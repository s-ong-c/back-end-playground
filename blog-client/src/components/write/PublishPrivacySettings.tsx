import * as React from 'react';
import styled, { css } from 'styled-components';
import PublishSection from './PublishSection';
import { GlobeIcon, LockIcon } from '../../static/svg';
import palette from '../../lib/styles/palette';

const PublishPrivacySettingsBlock = styled(PublishSection)`
  outline: none;
  border: none;
  .contents {
    display: flex;
  }
`;

const Button = styled.button<{ active: boolean }>`
  outline: none;
  border: none;
  flex: 1;
  height: 3rem;
  display: inline-flex;
  align-items: center;
  justify-content: flex-start;
  font-weight: bold;
  font-size: 1.125rem;
  box-shadow: 0 0 4px 0 rgba(0, 0, 0, 0.05);
  padding: 0;
  padding-left: 1rem;
  border: 1px solid ${palette.teal5};
  border-radius: 4px;
  cursor: pointer;
  &:hover {
    background: ${palette.gray0};
  }
  ${props =>
    props.active &&
    css`
      border: solid 1px ${palette.teal5};
      color: ${palette.teal5};
    `}
  svg {
    margin-right: 1.5rem;
  }
  & + & {
    margin-left: 1rem;
  }
`;
export enum PrivacySetting {
  PRIVATE = 'PRIVATE',
  PUBLIC = 'PUBLIC',
}

export interface PublishPrivacySettingsProps {
  selected: PrivacySetting;
  onSelect: (value: PrivacySetting) => any;
}

const PublishPrivacySettings: React.SFC<PublishPrivacySettingsProps> = ({
  selected,
  onSelect,
}) => {
  return (
    <PublishPrivacySettingsBlock title="공개 설정">
      <Button active={selected === PrivacySetting.PUBLIC}>
        <GlobeIcon />
        전체 공개
      </Button>
      <Button active={selected === PrivacySetting.PRIVATE}>
        <LockIcon />
        나만 보기
      </Button>
    </PublishPrivacySettingsBlock>
  );
};

export default PublishPrivacySettings;
