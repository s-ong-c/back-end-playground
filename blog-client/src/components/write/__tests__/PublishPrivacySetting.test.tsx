import * as React from 'react';
import { render } from 'react-testing-library';
import PublishPrivacySettings, {
  PublishPrivacySettingsProps,
  PrivacySetting,
} from '../PublishPrivacySettings';

describe('PublishPrivacySettings', () => {
  const setup = (props: Partial<PublishPrivacySettingsProps> = {}) => {
    const initialProps: PublishPrivacySettingsProps = {
      selected: PrivacySetting.PUBLIC,
      onSelect: () => {},
    };
    const utils = render(
      <PublishPrivacySettings {...initialProps} {...props} />,
    );
    const buttons = {
      public: utils.getByText('전체 공개'),
      private: utils.getByText('나만 보기'),
    };
    return {
      ...utils,
      buttons,
    };
  };
  it('renders property', () => {
    setup();
  });
  it('matches snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
  describe('shows active style', () => {
    it('PUBLIC', () => {
      const utils = setup();
      expect(getComputedStyle(utils.buttons.public).color).toBe(
        'rgb(32, 201, 151)',
      );
    });
    it('PRIVATE', () => {
      const utils = setup({ selected: PrivacySetting.PRIVATE });
      expect(getComputedStyle(utils.buttons.private).color).toBe(
        'rgb(32, 201, 151)',
      );
    });
  });
  // it('call onSelect fucntion onClick')
});
