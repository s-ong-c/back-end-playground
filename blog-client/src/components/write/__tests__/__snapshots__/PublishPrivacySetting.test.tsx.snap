// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublishPrivacySettings matches snapshot 1`] = `
<div>
  <section
    class="sc-bwzfXH difkFg sc-bdVaJa bfthOh"
  >
    <h3>
      공개 설정
    </h3>
    <div
      class="contents"
    >
      <button
        class="sc-htpNat jSLHvc"
      >
        <svg>
          icon-globe.svg
        </svg>
        전체 공개
      </button>
      <button
        class="sc-htpNat itBopW"
      >
        <svg>
          icon-lock.svg
        </svg>
        나만 보기
      </button>
    </div>
  </section>
</div>
`;
